### DISCUSSION TITLE
curious abot how Kubescape works!

### DISCUSSION THREAD

**Discussion by landbison1** [kubescape/kubescape#1733] [Q&A] [answered]
kubescape is open source, but I have a question because I don't understand something. kubescape uses opa and rego, so it fetches resource information, configuration information of components (API Server, etcd) in the form of JSON data, and maps them to OBJs that will go in as INPUT. But is this OBJ also JSON? Is it a direct comparison between JSON and the rules defined in the regolibrary? 

Host sensors also get the settings of components deployed on the same node as the Kubelet as JSON. Do they map this to a JSON called OBJ?
I see in your code that you also get the settings in kubelet.conf, is it possible to do a static manifest check of kubelet.conf?

Another thing is the static manifest check. If I have a local manifest yaml file, does this scan first take the manifest as input and convert it to JSON and map it?

I want to know how Kubescape works.

**Comment by Oshratn** [comment]
@landbison1 thanks for reaching out always happy to help.
Recently, @slashben was on a podcast describing Kubescape in depth. You can find it [here](https://www.youtube.com/live/8BrEeC6PiiY?feature=shared).

Furthemore, there is a community meeting today at 1pm UTC / 3pm CEST / 9am EDT / 6:30pm IST.
Here is the Zoom link :point_right: https://zoom.us/j/95174063585
The meeting is roughly divided into two:
Highlight any cool things that have happened in the project since the last meeting, as well as discuss any items of concern for the community.
Triage issues and unblock PRs
To track these items we have a [Google document](https://docs.google.com/document/d/1X_eyhPzJvb4ascVQ2e0jN87LAvq7lTuXT5d4gQxi8us/edit?usp=sharing) - if you have a highlight or other agenda item, please make a note in the document.

You are warmly invited to participate and get all your questions answered. If for some reason you can't make it, I have added this issue to today's agenda and you can then go back to the recording once posted.

**Comment by matthyx** [comment, marked as answer]
> kubescape is open source, but I have a question because I don't understand something. kubescape uses opa and rego, so it fetches resource information, configuration information of components (API Server, etcd) in the form of JSON data, and maps them to OBJs that will go in as INPUT. But is this OBJ also JSON? Is it a direct comparison between JSON and the rules defined in the regolibrary?
Yes.

> Host sensors also get the settings of components deployed on the same node as the Kubelet as JSON. Do they map this to a JSON called OBJ?
Yes https://github.com/kubescape/host-scanner/blob/master/sensor/kubelet.go#L32-L48

> I see in your code that you also get the settings in kubelet.conf, is it possible to do a static manifest check of kubelet.conf?
Possibly it would work.

> Another thing is the static manifest check. If I have a local manifest yaml file, does this scan first take the manifest as input and convert it to JSON and map it?
Yes

I hope I didn't write too many mistakes @Bezbran @dwertent

**Reply by Bezbran** [reply]
Yes, eventually all the inputs to the rego are being marshaled to JSON.
This is done by the OPA itself [here](https://github.com/open-policy-agent/opa/blob/3c2fc5da8530906440d3daee8ce72e97254f56b7/rego/rego.go#L1925-L1926).
